app/controllers/api/v1/admin/statuses_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Api::V1::Admin::StatusesController < Api::BaseController
+  # modeled on api/v1/admin/accounts_controller.rb
+
+  include Authorization
+  include AccountableConcern
+
+  before_action -> { authorize_if_got_token! :'admin:read', :'admin:read:statuses' }, only: [:show]
+  before_action -> { authorize_if_got_token! :'admin:write', :'admin:write:statuses' }, except: [:show]
+  before_action :set_status
+
+  after_action :verify_authorized
+
+  def show
+    authorize [:admin, @status], :show?
+    render json: @status, serializer: REST::StatusSerializer
+  end
+
+  def destroy
+    # modeled on handle_delete from status_batch_action.rb
+    authorize [:admin, @status], :destroy?
+    ApplicationRecord.transaction do
+      @status.discard_with_reblogs
+      log_action :destroy, @status
+      Tombstone.find_or_create_by(uri: @status.uri, account: @status.account)
+    end
+    json = render_to_body json: @status, serializer: REST::StatusSerializer, source_requested: true
+
+    RemovalWorker.perform_async(@status.id, { 'preserve' => @status.account.local?, 'immediate' => !@status.account.local? })
+
+    render json: json
+  end
+
+  def unsensitive
+    # modeled on undo_mark_statuses_as_sensitive from approve_appeal_service.rb
+    authorize [:admin, @status], :update?
+    representative_account = Account.representative
+    ApplicationRecord.transaction do
+      UpdateStatusService.new.call(@status, representative_account.id, sensitive: false) if @status.with_media?
```

Comment:
I'm curious about including `if @status.with_media?`. I see that's how they are doing it in the `approve_appeal_service` but I don't really get why. Doesn't this mean that `UpdateStatusService` won't get called if the status doesn't have media?

Reply:
Yeah, it's a bit weird but it does mean that `UpdateStatusService` won't get called if the status doesn't have media. I left it like this to match the equivalent action – `mark_as_sensitive` – in `status_batch_action.rb`, which similarly doesn't call `UpdateStatusService` with `sensitive: true` unless there is media.

Since this method is intended to undo moderation decisions, I figured it made sense to mirror the behavior.

Zooming out, I guess the intention is that moderators should only be able to make visual media as sensitive. I'll ask the T&S team if they want to change that policy to allow marking text-only posts as sensitive. Then, we can change both the action and undo action together.

Reply:
perfect, that sounds excellent!